client/src/crypto/cert.rs

```diff
@@ -1,25 +1,29 @@
-use anyhow::{bail, Context, Result};
 use rustls::{Certificate, PrivateKey, RootCertStore};
 use rustls_pemfile::{certs, pkcs8_private_keys, rsa_private_keys};
+use selium_std::errors::{Result, SeliumError};
 use std::{fs, path::Path};
 
 pub type KeyPair = (Vec<Certificate>, PrivateKey);
 
 fn load_key<T: AsRef<Path>>(path: T) -> Result<PrivateKey> {
     let path = path.as_ref();
-    let key = fs::read(path).context("failed to read private key")?;
+    let key =
+        fs::read(path).map_err(|_| SeliumError::InvalidKeys("failed to read private key."))?;
```

Comment:
Did you want to attach the upstream error to your error type? That way you have causation. E.g. was it a permissions issue, a network mount gone bad etc.

> The Error trait’s source() method is implemented to return whichever field has a #[source] attribute or is named source, if any. This is for identifying the underlying lower level error that caused your error.

```
#[derive(Error, Debug)]
pub struct MyError {
    msg: String,
    #[source]  // optional if field name is `source`
    source: anyhow::Error,
}
```